crates/sui-core/src/epoch/committee_store.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use rocksdb::Options;
+use std::path::PathBuf;
+use sui_storage::default_db_options;
+use sui_types::base_types::ObjectID;
+use sui_types::committee::{Committee, EpochId};
+use sui_types::error::{SuiError, SuiResult};
+use typed_store::rocks::DBMap;
+use typed_store::traits::TypedStoreDebug;
+
+use sui_types::fp_ensure;
+use typed_store::Map;
+use typed_store_derive::DBMapUtils;
+
+#[derive(DBMapUtils)]
+pub struct CommitteeStore {
+    /// Map from each epoch ID to the committee information.
+    /// TODO: We may also want to store the checkoint sequence number in each epoch that contains
+    /// the committee for the next epoch.
+    #[default_options_override_fn = "committee_table_default_config"]
+    pub(crate) committee_map: DBMap<EpochId, Committee>,
+}
+
+// These functions are used to initialize the DB tables
+fn committee_table_default_config() -> Options {
+    default_db_options(None, None).1
+}
+
+impl CommitteeStore {
+    pub fn new(path: PathBuf, genesis_committee: &Committee, db_options: Option<Options>) -> Self {
+        let committee_store = Self::open_tables_read_write(path, db_options, None);
+        if committee_store.database_is_empty() {
+            committee_store
+                .init_genesis_epoch(genesis_committee.clone())
```

Comment:
is genesis_epoch or a similar concept stored on chain today? e.g. when one reverse traverse the hashchain , how do they verify the genesis committee/state? 

There's certified genesis.blob but is it stored on chain?

Reply:
I think the assumption is that the genesis committee is the one thing that everyone has to trust through some offline methods (e.g. download from a link)